Orange/widgets/visualize/owvenndiagram.py

```diff
@@ -598,27 +594,29 @@ def create_from_rows(self, relevant_keys, relevant_ids, selection=False):
 
     def expand_table(self, table, atrs, metas, cv):
         exp = []
-        row_vals = {'attributes' : 'x', 'class_vars' : 'y', 'metas' : 'metas'}
-        l = 1 if isinstance(table, RowInstance) else len(table)
-        ids = table.id.reshape(-1,1) if isinstance(table, RowInstance) else table.ids.reshape(-1, 1)
-        atr_vals = row_vals if isinstance(table, RowInstance) else self.atr_vals
+        n = 1 if isinstance(table, RowInstance) else len(table)
+        if isinstance(table, RowInstance):
+            ids = table.id.reshape(-1, 1)
+            atr_vals = self.row_vals
+        else:
+            ids = table.ids.reshape(-1, 1)
+            atr_vals = self.atr_vals
         for all_el, atr_type in zip([atrs, metas, cv], self.atr_types):
             cur_el = getattr(table.domain, atr_type)
-            array = np.empty((l, len(all_el)))
-            array.fill(np.nan)
+            array = np.full((n, len(all_el)), np.nan)
             if cur_el:
                 perm = get_perm(cur_el, all_el)
                 b = getattr(table, atr_vals[atr_type])
                 array[:, perm] = b
             exp.append(array)
-        return exp[0], exp[1], exp[2], ids
+        return (*exp, ids)
 
     def extract_rowwise_duplicates(self, var_dict, ids, relevant_keys):
-        all_ids = sorted(list(reduce(set.union, [set(val.keys()) for val in ids.values()], set())))
-        sort_key = lambda var: var.name
-        all_atrs = sorted((var for var in var_dict['attributes'].keys()), key=sort_key)
-        all_metas = sorted((var for var in var_dict['metas'].keys()), key=sort_key)
-        all_cv = sorted((var for var in var_dict['class_vars'].keys()), key=sort_key)
+        all_ids = sorted(reduce(set.union, [set(val) for val in ids.values()], set()))
+        sort_key = attrgetter("name")
```

Comment:
Primarily to avoid complaints about assigning lambda functions to identifier (you'd be supposed to use `def` in that case).